trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,159 @@
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    if kernel_priors:
+        prior_scale = tf.cast(1.0, dtype=gpflow.default_float())
+        kernel.variance.prior = tfp.distributions.LogNormal(
+            tf.math.log(empirical_variance), prior_scale
+        )
+        kernel.lengthscales.prior = tfp.distributions.LogNormal(
+            tf.math.log(kernel.lengthscales), prior_scale
+        )
+
+    model = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    return model
+
+
+def build_sgpr(
+    data: Dataset,
+    search_space: Optional[Box] = None,
+    trainable_likelihood: bool = False,
+    trainable_inducing_points: bool = False,
+) -> SGPR:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+    num_data = len(data.observations)
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    if search_space:
+        inducing_points = search_space.sample_sobol(num_data)
+    else:
+        inducing_points = data.query_points[:num_data, :]
+
+    model = gpflow.models.SGPR(data.astuple(), kernel, inducing_points, noise_variance=1e-5)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    if not trainable_inducing_points:
+        gpflow.set_trainable(model.inducing_variable, False)
+
+    return model
+
+
+def build_vgp(data: Dataset, trainable_likelihood: bool = False) -> VGP:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    likelihood = gpflow.likelihoods.Gaussian(variance=1e-5)
+    model = gpflow.models.VGP(data.astuple(), kernel, likelihood)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    return model
+
+
+def build_vgp_classification(data: Dataset, trainable_likelihood: bool = False) -> VGP:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    likelihood = gpflow.likelihoods.Bernoulli()
+    model = gpflow.models.VGP(data.astuple(), kernel, likelihood)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    return model
+
+
+def build_svgp(
+    data: Dataset,
+    search_space: Optional[Box] = None,
+    trainable_likelihood: bool = False,
+    trainable_inducing_points: bool = False,
```

Comment:
number if inducing points is a necessary argument here

Reply:
optional likelihood argument, to allow using SVGP for classification?

Reply:
is it sensible to have more inducing points than number of data points?